app/controllers/v1/sessions_controller.rb

```diff
@@ -387,6 +389,11 @@ def invalid_message_timestamp_error?(message)
       message.match(FIM_INVALID_MESSAGE_TIMESTAMP)
     end
 
+    def set_user_client_id
```

Comment:
I think this is a bit weird, why not set this when we create the user identity object?

Reply:
I first looked into that, the user identity gets created entirely out of the SSOe SAML response given that we call it inside `UserSessionForm` at the start of the `user_login` method. The client id is coming from the SAMLTracker that we set ourselves, so I didn't see a way to include it in the user identity creation without injecting it into the `UserSessionForm`.

So it felt cleaner to tack it on at the end at the controller level where the tracker is available instead of messing with the way we invoke the `UserSessionForm`.

Reply:
It looks like in `user_attributes/ssoe.rb`, the `SAMLRequestTracker` is already retrieved and special things happen depending on the `application`, so I think you can just add the client_id: to `sign_in` in that file and you should be good to go